internal/controller/controller.go

```diff
@@ -72,39 +94,40 @@ func NewCaddyController(kubeClient *kubernetes.Clientset, restClient rest.Interf
 	controller.podInfo = podInfo
 
 	// load caddy config from file if mounted with config map
-	var caddyCfgMap *c.Config
-	cfgPath := "/etc/caddy/config.json"
-	if _, err := os.Stat(cfgPath); !os.IsNotExist(err) {
+	caddyCfgMap, err := loadCaddyConfigFile("/etc/caddy/config.json")
+	if err != nil {
+		logrus.Fatalf("Unexpected error reading config.json: %v", err)
+	}
+
+	if caddyCfgMap != nil {
 		controller.usingConfigMap = true
+	}
 
-		file, err := os.Open(cfgPath)
-		if err != nil {
-			log.Fatal(err)
-		}
-		defer file.Close()
+	// create 2 types of informers: one for the caddy NS and another one for ingress resources
+	ingInfFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, syncInterval, informers.WithNamespace(cfg.WatchNamespace))
```

Comment:
I prefer extended name (eg: `ingressInformerFactory` & `caddyInformerFactory`) to make it more understandable in the context (I know is controversial opinion, so I'll accept if you prefer to keep the name shorter :D )